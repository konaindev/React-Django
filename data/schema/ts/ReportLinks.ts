import * as t from "./types";

/** A named link to a single available project report */
interface ReportLink {
  /** The relative link to the project report page */
  url: string;

  /** A human-readable description of this specific report; can be ignored. */
  description: string;
}

/** The common structure used in all report pages to define available reports. */
export interface ReportLinks {
  /** The available baseline report for this project, if any. */
  baseline: ReportLink | null;

  /** The available performance reports for this project, if any. */
  performance: ReportLink[] | null;

  /** The available total addressable market (TAM) report for this project, if any. */
  market: ReportLink | null;

  /** The available modeling report for this project, if any. */
  modeling: ReportLink | null;
}
