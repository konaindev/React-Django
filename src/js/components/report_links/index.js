import React, { Component } from "react";
import PropTypes from "prop-types";
import cn from "classnames";
import { Link } from "react-router-dom";
import "./report_links.scss";

/**
 * @class ReportLinks
 *
 * @classdesc Renders the top-level report links for any report page.
 */
export default class ReportLinks extends Component {
  static propTypes = {
    current_report_name: PropTypes.string.isRequired,
    report_links: PropTypes.object.isRequired
  };

  renderLink(report_friendly_name, report_name, optional_report_link) {
    const names = cn(
      {
        selected: report_name == this.props.current_report_name
      },
      optional_report_link == null ? "disabled" : "enabled"
    );
    return (
      <li className={names} data-url={optional_report_link?.url || "#"}>
        <Link
          style={{ color: "inherit", textDecoration: "inherit" }}
          to={
            optional_report_link && optional_report_link.url
              ? optional_report_link.url
              : "#"
          }
        >
          {report_friendly_name}
        </Link>
      </li>
    );
  }

  render() {
    // TODO CONSIDER: report_links should probably contain friendly names, too?
    return (
      <div className="project-report-links">
        <ul>
          {this.renderLink(
            "Baseline",
            "baseline",
            this.props.report_links.baseline
          )}
          {this.props.report_links.market &&
            this.renderLink(
              "Market Analysis",
              "market",
              this.props.report_links.market
            )}
          {this.renderLink(
            "Modeling",
            "modeling",
            this.props.report_links.modeling
          )}
          {this.props.report_links.campaign_plan &&
            this.renderLink(
              "Campaign Plan",
              "campaign_plan",
              this.props.report_links.campaign_plan
            )}
          {this.renderLink(
            "Performance",
            "performance",
            this.props.report_links.performance?.[0]
          )}
        </ul>
      </div>
    );
  }
}
