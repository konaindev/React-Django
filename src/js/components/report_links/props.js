export const reportLinksAll = {
  baseline: {
    url: "/projects/pro_example/baseline/",
    description: "Baseline Period (Jan 01 2017 - May 07 2018)"
  },
  performance: [
    {
      url: "/projects/pro_example/performance/last-week/",
      description: "Last Week (Dec 24 2018 - Dec 31 2018)"
    }
  ],
  modeling: {
    url: "/projects/pro_example/modeling/",
    description: "Modeling"
  },
  campaign_plan: {
    url: "/projects/pro_example/campaign_plan/",
    description: "Campaign Plan"
  },
  market: {
    url: "/projects/pro_example/market/",
    description: "Total Addressable Market"
  }
};

export const reportLinksPartial = {
  ...reportLinksAll,
  modeling: null,
  campaign_plan: null,
  market: null
};

export const currentReportType = "baseline";

export default { currentReportType, reportLinks: reportLinksAll };
