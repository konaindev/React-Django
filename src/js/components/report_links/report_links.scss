@import "css/base.scss";

$link-h-spacing: 1.625rem; // 26px
$link-v-spacing: 0.875rem; // 14px

.project-report-links {
    display: inline-block;
}

.project-report-links ul {
    display: inline-block;
    list-style: none;
    padding: 0;
    margin-bottom: -1px;
    margin-left: -1 * $link-h-spacing;
}

.project-report-links li {
    cursor: pointer;
    display: inline-block;
    margin: 0 $link-h-spacing;
    padding: $link-v-spacing 0;
    color: $nav-link-color;
    font-size: $text-base;
    letter-spacing: 0.5px;
    line-height: $text-xl;
}

.project-report-links li.disabled {
    color: $nav-link-disabled-color;
}

.project-report-links li.selected {
    border-bottom: solid 1px $nav-link-active-border-color;
}

.project-report-links li.enabled:hover {
    color: $nav-link-hover-color;
}
