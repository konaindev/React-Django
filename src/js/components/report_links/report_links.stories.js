import React from "react";

import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import { linkTo } from "@storybook/addon-links";

import { report_links } from "../project_page/project_page.stories.js";

import ReportLinks from "./index";

const current_report_name = "baseline";

export const props = { current_report_name, report_links };

storiesOf("ReportLinks", module).add("default", () => (
  <ReportLinks {...props} />
));
