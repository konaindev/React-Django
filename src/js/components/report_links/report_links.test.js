import renderer from "react-test-renderer";
import ReportLinks from "./index";
import { props } from "./report_links.stories";

describe("ReportLinks", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ReportLinks {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
