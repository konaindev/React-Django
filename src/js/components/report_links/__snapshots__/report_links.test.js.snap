// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportLinks renders correctly 1`] = `
<div
  className="project-report-links"
>
  <ul>
    <li
      className="selected enabled"
      data-url="/projects/pro_example/baseline/"
      onClick={[Function]}
    >
      Baseline
    </li>
    <li
      className="enabled"
      data-url="/projects/pro_example/market/"
      onClick={[Function]}
    >
      Market Analysis
    </li>
    <li
      className="enabled"
      data-url="/projects/pro_example/modeling/"
      onClick={[Function]}
    >
      Modeling
    </li>
    <li
      className="enabled"
      data-url="/projects/pro_example/campaign_plan/"
      onClick={[Function]}
    >
      Campaign Plan
    </li>
    <li
      className="enabled"
      data-url="/projects/pro_example/performance/last-week/"
      onClick={[Function]}
    >
      Performance
    </li>
  </ul>
</div>
`;
